src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridItemModel.cs

```diff
@@ -88,59 +89,61 @@ public bool IsExpanded
         }
     }
 
-    private readonly IDisposable _modelActivationDisposable;
-
+    private readonly IDisposable _hasChildrenDisposable;
     private readonly SerialDisposable _childrenCollectionInitializationSerialDisposable = new();
     private readonly SerialDisposable _childrenObservableSerialDisposable = new();
 
-    protected TreeDataGridItemModel()
+    protected TreeDataGridItemModel(IObservable<bool>? hasChildrenObservable = null,
+        IObservable<IChangeSet<TModel, TKey>>? childrenObservable = null)
     {
+        HasChildrenObservable = hasChildrenObservable ?? Observable.Return(false);
+        ChildrenObservable = childrenObservable ?? Observable.Empty<IChangeSet<TModel, TKey>>();
         _childrenView = _children.ToNotifyCollectionChanged();
 
-        _modelActivationDisposable = WhenModelActivated(this, static (model, disposables) =>
+        // NOTE(sewer): If you're reading this during PR review; this code is marked
+        // for obsoletion as it will be replaced by `CompositeItemModel` system.
+        // So am not building anything more elaborate here.
```

Comment:
`TreeDataGridItemModel` is not marked as obsolete. It's still the base class for all things TreeDataGrid related as it facilitates logic for expanding and showing child item models.

The changes in this file should be reverted.